awx/main/scheduler/task_manager.py

```diff
@@ -255,6 +256,11 @@ def start_task(self, task, rampart_group, dependent_tasks=None, instance=None):
                              rampart_group.name, task.log_format))
                 return
 
+        # Before task leaves pending state, ensure that job_event partitions exist
+        # TODO: Currently, only creating partition for jobs. Drop contiditional once
+        # there are partitions for all job event types
+        if task_actual['type'] == 'job':
+            create_partition()
```

Comment:
I *do* think this is a reasonable place to put this function call, because it's within the scope of the global task manager advisory lock, and it's *right before* we anticipate we'll need the partition.

This doesn't need to happen now, but we should think about what it would mean if this function called *failed*.  How would an uncaught exception behave in this portion of the task manager?  Would we see an error?  Would the task manager exit gracefully and "try again" in 30s?  Is that okay?

cc @chrismeyersfsu 

This is a complicated and sensitive part of the overall system, so we should start pondering on it early 😄

Reply:
It might even make more sense to put in in the "post commit" function where we actually *are sure* we're about to run an `.apply_async()` call:

https://github.com/ansible/awx/pull/9341/files#diff-a37220424979b0075fa6e25bf8c309d671f30641e64830c27cd046272c73a703R333

Reply:
Sounds like we're also ready to call this function for _all_ event types, not just job events?

Reply:
yep, got pulled off before I was able to make this change 👍